src/Web/Microsoft.NET.Sdk.Web.Targets/Sdk.DefaultItems.targets

```diff
@@ -0,0 +1,39 @@
+<!--
+***********************************************************************************************
+Sdk.DefaultItems.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved.
+***********************************************************************************************
+-->
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Import Project="$(MSBuildThisFileDirectory)BundledVersions.props"
+		Condition="Exists('$(MSBuildThisFileDirectory)BundledVersions.props')" />
+
+  <Choose>
+    <When Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' AND '$(_TargetFrameworkVersionWithoutV)' == '2.1'" >
+      <PropertyGroup>
+        <AspNetCoreSharedFrameworkPackageName Condition="'$(AspNetCoreSharedFrameworkPackageName)' == ''">Microsoft.AspNetCore.App</AspNetCoreSharedFrameworkPackageName>
+        <AspNetCoreSharedFrameworkVersion Condition="'$(AspNetCoreSharedFrameworkVersion)' == ''">$(BundledMicrosoftAspNetCoreAppPackageVersion)</AspNetCoreSharedFrameworkVersion>
+      </PropertyGroup>
+
+      <ItemGroup>
+        <AspNetCoreAppPackageReference Include="@(PackageReference->WithMetadataValue('Identity', '$(AspNetCoreSharedFrameworkPackageName)'))" />
+        <ExplicitAspNetCoreAppPackageReference Include="@(AspNetCoreAppPackageReference->HasMetadata('Version'))" />
+
+        <PackageReference
+          Update="$(AspNetCoreSharedFrameworkPackageName)"
+          Condition="'@(AspNetCoreAppPackageReference->Count())' == '1' AND '@(ExplicitAspNetCoreAppPackageReference->Count())' == '0'">
+          <Version>$(AspNetCoreSharedFrameworkVersion)</Version>
+          <IsImplicitlyDefined>true</IsImplicitlyDefined>
+        </PackageReference>
+        <ExpectedPlatformPackages Include="$(AspNetCoreSharedFrameworkPackageName)" ExpectedVersion="$(AspNetCoreSharedFrameworkVersion)" />
```

Comment:
What is this item group for?

Reply:
It's to generate an error if you restore for one version of the reference but build with another one.  This would happen if you restore as framework dependent, and try to build for self-contained.  See https://github.com/dotnet/sdk/pull/2148 and https://github.com/dotnet/designs/pull/36.